tests/wpunit/PostConnectionPaginationTest.php

```diff
@@ -82,706 +87,587 @@ public function createPostObject( $args ) {
+		 */
 
-		// assert there are 2 items in the query
-		$this->assertCount( 2, $actual['data']['posts']['nodes'] );
+		// Set the variables to use in the GraphQL query.
+		$variables['after'] = $actual['data']['posts']['pageInfo']['endCursor'];
 
-		// Assert the first item is the 3rd most recent post
-		$this->assertSame( $this->created_post_ids[2], $actual['data']['posts']['nodes'][0]['databaseId'] );
+		// Set the variables to use in the WP query.
+		$query_args['offset'] = 2;
 
-		// Assert the 2nd item is the 4th most recent post
-		$this->assertSame( $this->created_post_ids[3], $actual['data']['posts']['nodes'][1]['databaseId'] );
+		// Run the GraphQL Query
+		$expected = $wp_query->query( $query_args );
 
-		// Query the next page
-		$actual = graphql( [
-			'query'     => $query,
-			'variables' => [
-				'first'  => 2,
-				'after'  => $actual['data']['posts']['pageInfo']['endCursor'],
-				'last'   => null,
-				'before' => null,
-			],
-		] );
+		$actual = $this->graphql( compact( 'query', 'variables' ) );
 
-		// assert there are 2 items in the query
-		$this->assertCount( 2, $actual['data']['posts']['nodes'] );
+		$this->assertValidPagination( $expected, $actual );
+		$this->assertEquals( true, $actual['data']['posts']['pageInfo']['hasPreviousPage'] );
+		$this->assertEquals( true, $actual['data']['posts']['pageInfo']['hasNextPage'] );
 
-		// Assert the first item is the 5th most recent post
-		$this->assertSame( $this->created_post_ids[4], $actual['data']['posts']['nodes'][0]['databaseId'] );
+		/**
+		 * Test the last two results.
+		 */
 
-		// Assert the 2nd item is the 6th most recent post
-		$this->assertSame( $this->created_post_ids[5], $actual['data']['posts']['nodes'][1]['databaseId'] );
+		// Set the variables to use in the GraphQL query.
+		$variables['after'] = $actual['data']['posts']['pageInfo']['endCursor'];
 
-		// Query the previous page
-		$actual = graphql( [
-			'query'     => $query,
-			'variables' => [
-				'first'  => null,
-				'after'  => null,
-				'last'   => 2,
-				'before' => $actual['data']['posts']['pageInfo']['startCursor'],
-			],
-		] );
+		// Set the variables to use in the WP query.
+		$query_args['offset'] = 4;
 
-		// assert there are 2 items in the query
-		$this->assertCount( 2, $actual['data']['posts']['nodes'] );
+		// Run the GraphQL Query
+		$expected = $wp_query->query( $query_args );
+		$actual   = $this->graphql( compact( 'query', 'variables' ) );
 
-		// Assert the first item is the 3rd most recent post
-		$this->assertSame( $this->created_post_ids[2], $actual['data']['posts']['nodes'][0]['databaseId'] );
+		$this->assertValidPagination( $expected, $actual );
+		$this->assertEquals( true, $actual['data']['posts']['pageInfo']['hasPreviousPage'] );
 
-		// Assert the 2nd item is the 4th most recent post
-		$this->assertSame( $this->created_post_ids[3], $actual['data']['posts']['nodes'][1]['databaseId'] );
+		if ( isset( $graphql_args['where']['search'] ) ) {
```

Comment:
@jasonbahl ☝️

Reply:
@justlevine looks like this was a false bug. 

I tried reproducing and couldn't but then realized that the test was creating 7 posts with `uniqueString` and then asserting that page 3 of 2 posts per page should have `hasNextPage` as `false`. But since there were 7 posts, `hasNextPage` was properly resolving as `true` as page 3 returns nodes 5 & 6, and page 4 would return the 7th node.

I updated the test to only create 6 posts and it's good now. 🙌🏻